streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -875,4 +935,25 @@ private void initializeTransactions() {
             );
         }
     }
+
+    static String encodeTimestamp(final long partitionTime) {
+        final ByteBuffer buffer = ByteBuffer.allocate(9);
+        buffer.put(LATEST_MAGIC_BYTE);
+        buffer.putLong(partitionTime);
+        return Base64.toBase64String(buffer.array());
+    }
+
+    static long decodeTimestamp(final String encryptedString) {
+        if (encryptedString.length() == 0) {
+            return -1;
+        }
+        final ByteBuffer buffer = ByteBuffer.wrap(Base64.decode(encryptedString));
+        final byte version = buffer.get();
+        switch (version) {
+            case LATEST_MAGIC_BYTE:
+                return buffer.getLong();
+            default: 
+                return -1;
```

Comment:
Same as above:

Also, we should log a WARN message there, that the the found metadata is corrupted and cannot be decoded.